pkg/reconciler/events/cloudevent/cloudeventsfakeclient.go

```diff
@@ -130,10 +107,33 @@ func (c *FakeClient) CheckCloudEventsUnordered(t *testing.T, testName string, wa
 			}
 		}
 		if !found {
-			t.Errorf("unexpected event received: %q", event)
+			t.Errorf("unexpected event received: %q", got)
 		}
 	}
-	if len(expected) != 0 {
-		t.Errorf("%d events %#v are not received", len(expected), expected)
+}
+
+// readEventsFromChannel returns a slice of all events sent on the channel.
+// It fails the test if the incorrect number of events were received.
+func readEventsFromChannel(t *testing.T, c chan string, numExpectedEvents int) []string {
+	var gotEvents []string
+	timeout := time.After(wait.ForeverTestTimeout)
+	// Read len(wantEvents) events from the channel, blocking until they have all been received
+	// or the test reaches its (long) timeout
+	for i := 0; i < numExpectedEvents; i++ {
+		select {
+		case got := <-c:
+			gotEvents = append(gotEvents, got)
+		case <-timeout:
+			t.Errorf("Expected event, got nothing")
```

Comment:
this may not be very accurate? Since we may receive some events?

Reply:
good point, fixed!